src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,299 @@
+    calling this function, a_list is not fully formed because the neighboring indicies of the
+    intersection points in b_list still need to be updated. We will have fully formed a_list
+    after calling _build_ab_list. Also at this point we still have not update the entry and
+    exit flags for a_list. The variables poly_a and poly_b are the original polygons that the
+    user passed in. The variables edges_a and edges_b are lists of the edges of each polygon, 
+    in the form of their cartesian points. Then intr_list is a list of the intersection points
+    in their cartesion point form. Then a_idx_list is a list of the indicies of where in a_list
+    an intersection point lies. The value at index i of a_idx_list is the location in a_list where
+    the ith point of intr_list lies. Then b_idx_list indicates which EDGE of poylgon b an intersection
+    point lies. Then alpha_a_list and alpha_b_list are vectors of scalar values which help order
+    the intersection points in the case that they lie on the same edge of a polygon. Finally,
+    a_list is an empty array of PolyNode objects that gets filled out when this function is run.
+    So is b_list. 
+"""
+function _build_a_list(intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list, a_list, b_list, poly_a, poly_b)
+    # Find intersection points and adds them to a_list
+    # "counter" is used to index all inter-related lists
+    counter = 1
+    # "acount" is used to index a_list
+    acount = 1
+
+    start = true
+    local p1
+    ii = 0
+    for p2 in GI.getpoint(poly_a)
+        if start
+            start = false
+        else
+        
+            # Add the first point of the edge to the list of points in a_list
+            if acount <= length(a_list)
+                a_list[acount] = PolyNode(ii, _tuple_point(p1), false, 0, false, 0.0)
+            else
+                push!(a_list, PolyNode(ii, _tuple_point(p1), false, 0, false, 0.0))
+            end
+            acount = acount + 1
+
+            jj = 0
+            start2 = true
+            local g1
+            for g2 in GI.getpoint(poly_b)
+                if start2
+                    start2 = false
+                else
+                    # Add the first point of the edge to b_list
+                    if ii == 1 
+                        # display("got here")
+                        # display(jj)
+                        # display(g1)
+                        b_list[jj] = PolyNode(jj, _tuple_point(g1), false, 0, false, 0.0)
+                    end
+
+                    # Check if edge jj of poly_b intersects with edge ii of poly_a
+                    # display(typeof([((p1[1], p1[2]), (p2[1], p2[2]))]))
+                    if _line_intersects([(_tuple_point(p1), _tuple_point(p2))], [(_tuple_point(g1), _tuple_point(g2))]);
+                        
+                        int_pt, alphas = _intersection_point((_tuple_point(p1), _tuple_point(p2)), (_tuple_point(g1), _tuple_point(g2)))
```

Comment:
You can just do _intersection_point and then check if the point is nothing for your if statement